contrib/scalar/scalar.c

```diff
@@ -191,9 +191,34 @@ static int toggle_maintenance(int enable)
 	return run_git("maintenance", enable ? "start" : "unregister", NULL);
 }
 
+static int add_or_remove_enlistment(int add)
```

Comment:
Does add_or_remove_enlistment() belong in this commit?  I wasn't expecting it here.

Reply:
Likewise for the changes in register_dir() and unregister_dir().

Reply:
You're right. I moved the respective code into the `register` and `unregister` commits.